src/Engine/Data/Texture.hpp

```diff
@@ -16,195 +17,181 @@ class Texture;
 namespace Ra {
 namespace Engine {
 namespace Data {
-/**
- * Describes the content and parameters of a texture.
+
+struct SamplerParameters {
+    /// OpenGL wrap mode in the s direction
+    GLenum wrapS { GL_CLAMP_TO_EDGE };
+    /// OpenGL wrap mode in the t direction
+    GLenum wrapT { GL_CLAMP_TO_EDGE };
+    /// OpenGL wrap mode in the r direction
+    GLenum wrapR { GL_CLAMP_TO_EDGE };
+    /// OpenGL minification filter ( GL_LINEAR or GL_NEAREST or GL_XXX_MIPMAP_YYY )
+    GLenum minFilter { GL_LINEAR };
+    /// OpenGL magnification filter ( GL_LINEAR or GL_NEAREST )
+    GLenum magFilter { GL_LINEAR };
+};
+
+struct ImageParameters {
+    /// Types for texels variant
+    using ImageType   = std::shared_ptr<void>;
+    using CubeMapType = std::array<std::shared_ptr<void>, 6>;
+
+    /// check which type is held by texels
+    template <typename TexelType>
+    bool isTexelOfType() const {
+        return std::holds_alternative<TexelType>( texels );
+    }
+
```

Comment:
I think the two following functions could be helpful
```suggestion
template<typename ContentType>
ContentType getContent() const {
    CORE_ASSERT( isTexelOfType<ContentType>(), "texture variant does not hold the requested type." );
    return std::get<ContentType>( texels );
}

template<typename ContentType>
void setContent( ContentType image ) { texels = image; }
```
and then {get|set}Image and {get|set}CubeMap should be simple wrapper (syntactic sugar) over these two.
This should enable client to manage texture in a templated way without testing explicitly the type of the texture.

Reply:
Since data is public, why don't just use
`std::get<ContentType>(imageParam.texels)`
?

Reply:
Let's delay to a use case, if needed we can add those get / set.